exporters/metric/cortex/auth.go

```diff
@@ -88,3 +92,99 @@ func (e *Exporter) addBearerTokenAuth(req *http.Request) error {
 
 	return nil
 }
+
+// buildClient returns a http client that uses TLS and has the user-specified proxy and
+// timeout.
+func (e *Exporter) buildClient() (*http.Client, error) {
+	// Create a TLS Config struct for use in a custom HTTP Transport.
+	tlsConfig, err := e.buildTLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	transport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+
+	// Convert proxy url to proxy function for use in the created Transport.
+	if e.config.ProxyURL != "" {
+		proxyURL, err := url.Parse(e.config.ProxyURL)
+		if err != nil {
+			return nil, err
+		}
+		proxy := http.ProxyURL(proxyURL)
+		transport.Proxy = proxy
+	}
```

Comment:
Can we add a default environment variable handling here for proxy specification since we are overriding the default transport: https://golang.org/pkg/net/http/#ProxyFromEnvironment